app/models/concerns/foreman_puppet/extensions/provisioning_template.rb

```diff
@@ -46,7 +46,7 @@ def templates_by_template_combinations(templates, hosts_or_conditions)
 
         def template_includes
           includes = super
-          tc_include = includes.detect { |i| i.key?(:template_combinations) }
+          tc_include = includes.detect { |i| i.is_a?(Hash) && i.key?(:template_combinations) }
           tc_include ||= includes << {}
           tc_include[:template_combinations] = %i[hostgroup environment]
           includes
```

Comment:
@ezr-ondrej: Does this make sense or should this be `tc_includes` as a return value.

Reply:
`tc_include` is a hash inside of the includes array. if it's not found, its added to the includes array on line 50.
So it is correct as is

Reply:
Ok. Makes sense. Thanks for the clarification!